File: src/main/java/no/fint/featuretoggle/FeatureToggleController.java
package no.fint.featuretoggle;

import no.finn.unleash.DefaultUnleash;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/api")
public class FeatureToggleController {

    @Value("${spring.application.name}")
    private String applicationName;

    private final DefaultUnleash unleashClient;

    public FeatureToggleController(DefaultUnleash unleashClient) {
        this.unleashClient = unleashClient;
    }

    @GetMapping("feature")
    public ResponseEntity<Map<String, Boolean>> getFeatures() {

        return ResponseEntity.ok(unleashClient
                .getFeatureToggleNames()
                .stream()
                .filter(feature -> feature.startsWith(applicationName))
                .collect(Collectors.toMap(feature -> stripApplicationName(applicationName, feature), unleashClient::isEnabled)));
    }

    private String stripApplicationName(String applicationName, String feature) {
        return feature.replace(applicationName + ".", "");
    }
}

File: src/main/kotlin/no/fint/featuretoggle/FeatureToggleController.kt
package no.fint.featuretoggle

import io.getunleash.Unleash
import org.springframework.beans.factory.annotation.Value
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class FeatureToggleController(
    @Value("\${spring.application.name}")
    private val applicationName: String,
    private val unleashClient: Unleash,
) {
    @GetMapping("/feature")
    fun feature(): ResponseEntity<Map<String, Boolean>> {
        val features =
            unleashClient.more().featureToggleNames
                .filter { it.startsWith(applicationName) }
                .associateBy({ it.replace("$applicationName.", "") }, { unleashClient.isEnabled(it) })
        return ResponseEntity.ok().body(features)
    }
}
